graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/messages/RemoveFieldsByValue.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.pipelineprocessor.functions.messages;
+
+import com.google.common.collect.ImmutableList;
+import org.graylog.plugins.pipelineprocessor.EvaluationContext;
+import org.graylog.plugins.pipelineprocessor.ast.functions.AbstractFunction;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionArgs;
+import org.graylog.plugins.pipelineprocessor.ast.functions.FunctionDescriptor;
+import org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor;
+import org.graylog.plugins.pipelineprocessor.rulebuilder.RuleBuilderFunctionGroup;
+import org.graylog2.plugin.Message;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.type;
+
+public class RemoveFieldsByValue extends AbstractFunction<Void> {
+    public static final String NAME = "remove_fields_by_value";
+    private static final String VALUES_ARG = "values";
+    private static final String PATTERN_ARG = "pattern";
+    private final ParameterDescriptor<String, Pattern> regexParam;
+    private final ParameterDescriptor<List, List> valuesParam;
+    private final ParameterDescriptor<Message, Message> messageParam;
+
+    public RemoveFieldsByValue() {
+        regexParam = ParameterDescriptor.string(PATTERN_ARG, Pattern.class)
+                .optional()
+                .transform(Pattern::compile)
+                .description("A regex specifying field values to be removed.").build();
+        valuesParam = type(VALUES_ARG, List.class).optional().description("A list of exact values of fields to be removed.").build();
```

Comment:
There might be some benefit to adding a case-insensitive values option, just in case values need to be removed where the case is uncertain.

Reply:
Case-insensitive matching can be accomplished with a regex using the `pattern` param if need be. I have a feeling this will typically be used to remove things like empty strings, dashes, or some known value like `<empty>`. Anything more complicated than exact matching I think is okay to handle in regex. That should keep the params for this function fairly simple. @danotorrey